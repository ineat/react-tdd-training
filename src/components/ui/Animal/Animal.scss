.animal-container {
  position: relative;
  cursor: pointer;

  &:first-of-type {
    margin-right: 2rem;
  }

  &:last-of-type {
    margin-left: 2rem;
  }

  .description {
    position: absolute;
    bottom: -20px;
    left: 0;
    z-index: 1;
    display: block;
    width: 350px;
    margin: 0;
    padding: 0;
    color: #444;
    font-size: 25px;
    font-weight: bold;
    text-decoration: none;
    text-align: center;
    -webkit-transition: .3s ease-in-out;
    transition: .3s ease-in-out;
    opacity: 0;
  }

  .animal {
    width: 350px;
    border-radius: 30px;

    -webkit-filter: sepia(100%);
    filter: sepia(100%);
    -webkit-transition: .3s ease-in-out;
    transition: .3s ease-in-out;

    &:hover {
      -webkit-filter: sepia(0);
      filter: sepia(0);
    }
    &:hover+span {
      bottom: 25px;
      opacity: 1;
    }
  }
}