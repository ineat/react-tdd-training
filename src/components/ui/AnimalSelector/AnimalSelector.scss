.animal-selector {
  display: inline-flex;
}