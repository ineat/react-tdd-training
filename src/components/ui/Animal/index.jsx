import * as React from 'react';

import './Animal.scss';

export default ({ image, alt, description }) => (
  <div className="animal-container">
    <img className="animal" src={image} alt={alt}/>
    <span className="description">
      {description}
    </span>
  </div>
);