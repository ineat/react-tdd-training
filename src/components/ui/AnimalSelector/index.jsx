import * as React from 'react';

import dogLogo from '../../../assets/images/dog.png';
import catLogo from '../../../assets/images/cat.png';

import './AnimalSelector.scss';
import Animal from '../Animal';

export default () => (
  <div className="animal-selector">
    <Animal image={catLogo} alt="Meow!" description="I love cats"/>
    <Animal image={dogLogo} alt="Wouf!" description="I love dogs"/>
  </div>
);