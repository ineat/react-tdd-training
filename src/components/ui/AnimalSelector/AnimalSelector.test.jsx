import * as React from 'react';
import { shallow, mount } from 'enzyme';

import AnimalSelector from '.';
import Animal from '../Animal';

describe('Animal selector', () => {
  const props = { onSelectAnimal: jest.fn() };
  it('Should render.', () => {
    expect(mount(<AnimalSelector {...props} />)).toMatchSnapshot();
  });

  it('Should call "onSelectAnimal" with "cat" type for first image.', () => {
    const wrapper = shallow(<AnimalSelector {...props} />);
    wrapper.find(Animal).at(0).prop('onSelectAnimal')();

    expect(props.onSelectAnimal).toHaveBeenCalledWith('cat');
  });

  it('Should call "onSelectAnimal" with "dog" type for second image.', () => {
    const wrapper = shallow(<AnimalSelector {...props} />);
    wrapper.find(Animal).at(1).prop('onSelectAnimal')();

    expect(props.onSelectAnimal).toHaveBeenCalledWith('dog');
  });
});