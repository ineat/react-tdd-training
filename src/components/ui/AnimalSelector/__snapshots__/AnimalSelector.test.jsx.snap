// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal selector Should render. 1`] = `
<_default
  onSelectAnimal={[MockFunction]}
>
  <div
    className="animal-selector"
  >
    <_default
      alt="Meow!"
      description="I love cats"
      image="cat.png"
      onSelectAnimal={[Function]}
    >
      <div
        className="animal-container"
      >
        <img
          alt="Meow!"
          className="animal"
          onClick={[Function]}
          src="cat.png"
        />
        <span
          className="description"
        >
          I love cats
        </span>
      </div>
    </_default>
    <_default
      alt="Wouf!"
      description="I love dogs"
      image="dog.png"
      onSelectAnimal={[Function]}
    >
      <div
        className="animal-container"
      >
        <img
          alt="Wouf!"
          className="animal"
          onClick={[Function]}
          src="dog.png"
        />
        <span
          className="description"
        >
          I love dogs
        </span>
      </div>
    </_default>
  </div>
</_default>
`;
